symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/ProviderLoader.java

```diff
@@ -0,0 +1,43 @@
+package com.symphony.bdk.core.util;
+
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiClientBuilderProvider;
+
+import lombok.Generated;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@API(status = API.Status.INTERNAL)
+@Slf4j
+@Generated
+public class ProviderLoader {
+
+  /**
+   * Load {@link ApiClient} implementation class using {@link ServiceLoader}.
+   *
+   * @return an {@link ApiClientBuilderProvider}.
+   */
+  public static ApiClientBuilderProvider findApiClientBuilderProvider() {
```

Comment:
Now you refactored it as a utility class, could we try to  make it a bit more generic? `public static <T> T lookupSingleService(Class<T> clz)`